katka/serializers.py

```diff
@@ -66,4 +61,20 @@ def validate(self, attrs):
         if not self.context['request'].user.groups.filter(name=attrs['team'].group.name).exists():
             raise PermissionDenied('User is not a member of this group')
 
-        return attrs
+        return super().validate(attrs)
+
+
+class ProjectSerializer(TeamChildMixin, serializers.ModelSerializer):
+    team = EmbeddedTeamSerializer(required=False, read_only=True)
```

Comment:
What if we did not include the team but just the public_identifier of the team? So resources don't include related resources. I would assume that the client would also have requested team resource itself?

Reply:
We talked about having URLs accessible in the FE based on slug, but for external resources (think the link between Deployments and Applications), you probably want to have a public_identifier.

This means we decided for now to expose both, otherwise you can not directly link to a resource without having to do a lookup (FE gets only slug, but you can only lookup based on public_identifier, you need to lookup one from the other).

So if the FE just gets a link (imagine deeplink) with just slugs, you need to get a list of teams, iterate through it to get the matching public_identifier, etc).

I would prefer to solve this differently that this though. Do you have a nicer solution maybe?

Reply:
I think it would be nicer to do everything based on public identifiers. That way you do not need the additional look ups and you are always referring to the same thing (and you never have to worry about uniqueness of a value either between teams). I am also thinking about performance here as now you need to do additional look ups in the database and while this might work for the 'core', this is really not something you would want for items outside of the core (since that would result in a request to the core).

Reply:
discussed offline, I'll remove it for now and we'll revisit later if we want to have slugs.

Reply:
I realize that the name is misleading: it's not meant to embed the whole other resource, just the identifying attributes (public_identifier and slug). I left it in as before, but I made the unique_together because that was missing. I'm open to changing the name of the serializer though.